netlogo-core/src/main/api/ModelLoader.scala

```diff
@@ -1,14 +1,35 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
 
 package org.nlogo.api
 
-import java.io.FileNotFoundException
+import java.io.{ FileNotFoundException, Writer }
 import java.net.URI
 import java.nio.file.{ Files, Paths }
 
 import org.nlogo.core.{ I18N, Model }
 
-import scala.util.{ Failure, Try }
 import scala.reflect.ClassTag
+import scala.util.{ Failure, Try }
+
+trait AbstractModelLoader {
+  def readModel(uri: URI): Try[Model]
+  def readModel(source: String, extension: String): Try[Model]
+  def save(model: Model, uri: URI): Try[URI]
+  def sourceString(model: Model, extension: String): Try[String]
+  def emptyModel(extension: String): Model
+  // these next two allow ManagerDialog to use the correct format for experiment loading/saving (IB 8/17/24)
```

Comment:
Please sign your comments "Isaac B" instead of "IB".  Our policy changed on this (particularly after Jeremy and I had conflicting initials).